src/main/host/syscall_handler.c

```diff
@@ -562,7 +562,8 @@ SysCallReturn syscallhandler_make_syscall(SysCallHandler* sys,
         SysCallCondition* condition = scr.cond;
         utility_assert(condition);
         syscallcondition_unref(condition);
-        scr = (SysCallReturn){.state = SYSCALL_DONE, .retval = -EINTR};
+        scr = (SysCallReturn){
+            .state = SYSCALL_DONE, .retval = -EINTR, .restartable = scr.restartable};
```

Comment:
The comment above says "The shim-side code will run the signal handlers before returning EINTR to the original syscall". This should be updated to include comments about what happens if it's restartable.